proto/connector_service.proto

```diff
@@ -27,6 +27,7 @@ message SinkConfig {
   string connector_type = 1;
   map<string, string> properties = 2;
   TableSchema table_schema = 3;
+  uint64 sink_id = 4;
```

Comment:
Instead of putting the `sink_id` in the `SinkConfig`, I think we can specify the sink_id only in the start_sink request. Otherwise, we will have to pass the sink id in the validation rpc like what we did in this PR. However, sink id is not relevant to the validation.